libsolidity/interface/CompilerStack.cpp

```diff
@@ -1251,14 +1264,47 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	tie(compiledContract.yulIR, compiledContract.yulIROptimized) = generator.run(_contract, otherYulSources);
 }
 
+void CompilerStack::generateEVM(ContractDefinition const& _contract)
+{
+	solAssert(m_stackState >= AnalysisPerformed, "");
+	if (m_hasError)
+		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Called generateEVM with errors."));
+
+	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
+	// Empty contract, ignore.
+	// TODO: check if this could happen in other cases.
+	if (compiledContract.yulIROptimized.empty())
```

Comment:
Hm, instead of this perhaps `if (!_contract.canBeDeployed()) return` is a better solution.